src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -680,7 +687,7 @@ private BoundExpression BindTupleExpression(TupleExpressionSyntax node, Diagnost
                         elementNames = ArrayBuilder<string>.GetInstance(arguments.Count);
                         for (int j = 0; j < i; j++)
                         {
-                            elementNames.Add(name ?? "Item" + j);
+                            elementNames.Add(name ?? TupleTypeSymbol.UnderlyingMemberName(j));
```

Comment:
The name will be visible through the symbol. It should be possible to get it even in presence of errors.